pyscf/mcpdft/otfnal.py

```diff
@@ -19,19 +19,110 @@
 from scipy import linalg
 from pyscf import lib, dft
 from pyscf.lib import logger
+from pyscf.dft2 import libxc
 from pyscf.dft.gen_grid import Grids
 from pyscf.dft.numint import _NumInt, NumInt
 from pyscf.mcpdft import pdft_veff, tfnal_derivs, _libxc, _dms, pdft_feff, pdft_eff
 from pyscf.mcpdft.otpd import get_ontop_pair_density
 from pyscf import __config__
 
+# Points to the libxc of the dft2 module
+dft.numint.libxc = libxc
+dft.numint.LibXCMixin.libxc = libxc
```

Comment:
This is still really impure: anything that happens after the `otfnal` module is loaded for the first time will use this modified, "monkey-patched" version of the PySCF core library, even if it's completely unrelated to MC-PDFT. As @Dayou-Zhang said on the previous commit, it's only the otfnal's `_numint` member that actually needs to see this:

```
self._numint.libxc = libxc
```
after lines 298 and 674.

Reply:
Done.